kythe/typescript/kythe.ts

```diff
@@ -0,0 +1,206 @@
+ *  - `signature`: a unique, opaque signature for a node
+ *  - `corpus`: a collection of related files the node is defined in
+ *  - `root`: a label denoting a distinct subset of the corpus
+ *  - `path`: the relative path of the file containing the node
+ *  - `language`: programming language the node belongs to
+ */
+export interface VName {
+  signature: string;
+  corpus: string;
+  root: string;
+  path: string;
+  language: string;
+}
+
+/**
+ * Kythe edge kinds. See
+ *   https://kythe.io/docs/schema/#_edge_kinds
+ *   https://github.com/kythe/kythe/tree/master/kythe/data/schema_index.textproto#L17
+ */
+export enum EdgeKind {
+  ALIASES = '/kythe/edge/aliases',
+  ALIASES_ROOT = '/kythe/edge/aliases/root',
+  ANNOTATED_BY = '/kythe/edge/annotatedby',
+  BOUNDED_LOWER = '/kythe/edge/bounded/lower',
+  BOUNDED_UPPER = '/kythe/edge/bounded/upper',
+  CHILD_OF = '/kythe/edge/childof',
+  CHILD_OF_CONTEXT = '/kythe/edge/childof/context',
+  COMPLETES = '/kythe/edge/completes',
+  COMPLETES_UNIQUELY = '/kythe/edge/completes/uniquely',
+  DEFINES = '/kythe/edge/defines',
+  DEFINES_BINDING = '/kythe/edge/defines/binding',
+  DEPENDS = '/kythe/edge/depends',
+  DOCUMENTS = '/kythe/edge/documents',
+  EXPORTS = '/kythe/edge/exports',
+  EXTENDS = '/kythe/edge/extends',
+  GENERATES = '/kythe/edge/generates',
+  INSTANTIATES = '/kythe/edge/instantiates',
+  INSTANTIATES_SPECULATIVE = '/kythe/edge/instantiates/speculative',
+  IMPUTES = '/kythe/edge/imputes',
+  NAMED = '/kythe/edge/named',
+  OVERRIDES = '/kythe/edge/overrides',
+  OVERRIDES_ROOT = '/kythe/edge/overrides/root',
+  OVERRIDES_TRANSITIVE = '/kythe/edge/overrides/transitive',
+  PARAM = '/kythe/edge/param',
+  REF = '/kythe/edge/ref',
+  REF_IMPLICIT = '/kythe/edge/ref/implicit',
+  REF_CALL = '/kythe/edge/ref/call',
+  REF_CALL_IMPLICIT = '/kythe/edge/ref/call/implicit',
+  REF_DOC = '/kythe/edge/ref/doc',
+  REF_EXPANDS = '/kythe/edge/ref/expands',
+  REF_EXPANDS_TRANSITIVE = '/kythe/edge/ref/expands/transitive',
+  REF_FILE = '/kythe/edge/ref/file',
+  REF_IMPORTS = '/kythe/edge/ref/imports',
+  REF_INCLUDES = '/kythe/edge/ref/includes',
+  REF_INIT = '/kythe/edge/ref/init',
+  REF_INIT_IMPLICIT = '/kythe/edge/ref/init/implicit',
+  REF_QUERIES = '/kythe/edge/ref/queries',
+  SATISFIES = '/kythe/edge/satisfies',
+  SPECIALIZES = '/kythe/edge/specializes',
+  SPECIALIZES_SPECULATIVE = '/kythe/edge/specializes/speculative',
+  TAGGED = '/kythe/edge/tagged',
+  TYPED = '/kythe/edge/typed',
+  UNDEFINES = '/kythe/edge/undefines',
+}
+
+/**
+ * A Kythe ordinal edge has the form of
+ *   `${EdgeKind}.${number}`
+ * This is represented as a branded string that is incompatible with a string
+ * but can be compared to a string.
+ */
+export type OrdinalEdge = string&{
+  __ordinalBrand: 'ordinal';
+};
+export function makeOrdinalEdge(edge: EdgeKind, ordinal: number): OrdinalEdge {
+  const ordinalEdge = `${edge}.${ordinal}`;
+  return <OrdinalEdge>ordinalEdge;
```

Comment:
I think
return ordinalEdge as OrdinalEdge;
The brackets syntax is obsolete

Reply:
Done.